<a href="https://colab.research.google.com/github/vitorjensen/Python_para_Dados/blob/main/Projetos/An%C3%A1lise%20Explorat%C3%B3ria%20de%20Carros/An%C3%A1lise_Explorat%C3%B3ria_de_Carros.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#### Realizando a análise exploratória de Automóveis.

#### Importação de bibliotecas

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
df = pd.read_csv('/content/drive/MyDrive/automoveis_dataset.csv')

In [4]:
# Apresentando os dados através das 5 primeiras linhas
df.head()

,Unnamed: 0,car_name,brand,model,vehicle_age,km_driven,seller_type,fuel_type,transmission_type,mileage,engine,max_power,seats,selling_price
0,0,Maruti Alto,Maruti,Alto,9,120000,Individual,Petrol,Manual,19.70,796,46.30,5,120000
1,1,Hyundai Grand,Hyundai,Grand,5,20000,Individual,Petrol,Manual,18.90,1197,82.00,5,550000
2,2,Hyundai i20,Hyundai,i20,11,60000,Individual,Petrol,Manual,17.00,1197,80.00,5,215000
3,3,Maruti Alto,Maruti,Alto,9,37000,Individual,Petrol,Manual,20.92,998,67.10,5,226000
4,4,Ford Ecosport,Ford,Ecosport,6,30000,Dealer,Diesel,Manual,22.77,1498,98.59,5,570000


##### Iniciando um tratamento de dados (Removendo colunas desnecessárias da base)

In [19]:
df.drop('Unnamed: 0', axis=1, inplace=True)

In [ ]:
# Reexibindo os dados (top)
df.head()

In [ ]:
# Exibindo os final do arquivo (bottom)
df.tail()

##### Exibindo as colunas do arquivo

In [9]:
df.columns

Index(['car_name', 'brand', 'model', 'vehicle_age', 'km_driven', 'seller_type',
       'fuel_type', 'transmission_type', 'mileage', 'engine', 'max_power',
       'seats', 'selling_price'],
      dtype='object')

##### Verificação dos Dados:
  - Verificação dos Dados **nulos** e **Tipos de Dados** nas colunas
  - Verificação de Dados **duplicados**
  - Verificação de Dados **únicos** por colunas

In [11]:
# Dados nulos e tipos de dados
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15411 entries, 0 to 15410
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   car_name           15411 non-null  object 
 1   brand              15411 non-null  object 
 2   model              15411 non-null  object 
 3   vehicle_age        15411 non-null  int64  
 4   km_driven          15411 non-null  int64  
 5   seller_type        15411 non-null  object 
 6   fuel_type          15411 non-null  object 
 7   transmission_type  15411 non-null  object 
 8   mileage            15411 non-null  float64
 9   engine             15411 non-null  int64  
 10  max_power          15411 non-null  float64
 11  seats              15411 non-null  int64  
 12  selling_price      15411 non-null  int64  
dtypes: float64(2), int64(5), object(6)
memory usage: 1.5+ MB


In [6]:
# Soma de Dados duplicados
df.duplicated().sum()

np.int64(0)

In [ ]:
# Valores únicos em cada coluna
df.nunique()

##### Resumo
  - 15411 linhas e 13 colunas
  - Não possui colunas nulas (Dados Faltantes)
  - Tipos de colunas: 6 categóricas e 7 númericas
  - Possui 167 linhas duplicadas

#####Segunda parte da Análise Exploratória:
  - Explorando Dados
  - Análise Univariada
      - Gráfico de frequência
  - Análise Múltivariada
      - Correlação
      - Média Bivariada

##### Análise Exploratória

In [11]:
# Print de dados únicos em colunas específicas
print("Categoria em 'seller_type:'",end=" ")
print(df['seller_type'].unique())
print("\n")

print("Categorias em 'fuel_type:'", end=" ")
print(df['fuel_type'].unique())
print("\n")

print("Categorias em 'transmission_type:'", end=" ")
print(df['transmission_type'].unique())
print("\n")

print("Categorias em 'seats:'", end=" ")
print(df['seats'].unique())
print("\n")

Categoria em 'seller_type:' ['Individual' 'Dealer' 'Trustmark Dealer']


Categorias em 'fuel_type:' ['Petrol' 'Diesel' 'CNG' 'LPG' 'Electric']


Categorias em 'transmission_type:' ['Manual' 'Automatic']


Categorias em 'seats:' [5 8 7 6 4 2 9 0]




##### Avaliando Possíveis inconsistências de Dados

In [20]:
# Checagem das colunas e possíveis erros.
df[df['seats'] == 0]

,car_name,brand,model,vehicle_age,km_driven,seller_type,fuel_type,transmission_type,mileage,engine,max_power,seats,selling_price
3217,Honda City,Honda,City,18,40000,Individual,Petrol,Manual,13.00,1493,100.00,0,115000
12619,Nissan Kicks,Nissan,Kicks,2,10000,Individual,Diesel,Manual,19.39,1461,108.49,0,1154000


In [26]:
# Reescrevendo os valores de 0 presentes nas linhas da coluna "seats" para 5
df.loc[df['seats'] == 0, 'seats'] = 5

# Ou se desejar remover as linhas contendo zero, use:
# df.drop(df[df['seats'] == 0].index, inplace=True)

In [ ]:
df[df['seats'] == 0]

In [30]:
# Removendo a coluna redundante "mileage"
df.drop('mileage', axis=1, inplace=True)

In [35]:
# Renomear a coluna "Km_Driven" para "mileage"
df.rename(columns={'km_driven': 'MileAge'}, inplace=True)

In [37]:
df.head(3)

,car_name,brand,model,vehicle_age,MileAge,seller_type,fuel_type,transmission_type,engine,max_power,seats,selling_price
0,Maruti Alto,Maruti,Alto,9,120000,Individual,Petrol,Manual,796,46.3,5,120000
1,Hyundai Grand,Hyundai,Grand,5,20000,Individual,Petrol,Manual,1197,82.0,5,550000
2,Hyundai i20,Hyundai,i20,11,60000,Individual,Petrol,Manual,1197,80.0,5,215000
